packages/adapters/src/adapters/Cornerstone3D/RTStruct/generateROIContourFromSegmentation.js

```diff
@@ -0,0 +1,203 @@
+                            contours,
+                            reducedSet.points
+                        );
+
+                    console.log(JSON.parse(JSON.stringify(processedContours)));
+
+                    /**
+                     * addContour - Adds a new ROI with related contours to ROIContourSequence
+                     *
+                     * @param {Object} newContour cornerstoneTools `ROIContour` object
+                     *
+                     * newContour = {
+                     *   name: string,
+                     *   description: string,
+                     *   contourSequence: array[contour]
+                     * }
+                     *
+                     * contour = {
+                     *   ContourImageSequence: array[
+                     *       { ReferencedSOPClassUID: string, ReferencedSOPInstanceUID: string}
+                     *     ]
+                     *   ContourGeometricType: string,
+                     *   NumberOfContourPoints: number,
+                     *   ContourData: array[number]
+                     * }
+                     */
+                    // Note: change needed if support non-planar contour representation is needed
+                    const sopCommon = metadataProvider.get(
+                        "sopCommonModule",
+                        imageVol.imageIds[i]
+                    );
+                    const ReferencedSOPClassUID = sopCommon.sopClassUID;
+                    const ReferencedSOPInstanceUID = sopCommon.sopInstanceUID;
+                    const ContourImageSequence = [
+                        { ReferencedSOPClassUID, ReferencedSOPInstanceUID } // NOTE: replace in dcmjs?
+                    ];
+                    contours.forEach((contour, index) => {
+                        const ContourGeometricType = contour.type;
+                        const NumberOfContourPoints =
+                            contour.contourPoints.length;
+                        const ContourData = [];
+
+                        contour.contourPoints.forEach(point => {
+                            const pointData = reducedSet.points[point];
+                            //pointData[0] = +pointData[0].toFixed(2);
+                            //pointData[1] = +pointData[1].toFixed(2);
+                            //pointData[2] = +pointData[2].toFixed(2);
+                            ContourData.push(pointData[0]);
+                            ContourData.push(pointData[1]);
+                            ContourData.push(pointData[2]);
+                        });
+
+                        contourSequence.push({
+                            ContourImageSequence,
+                            ContourGeometricType,
+                            NumberOfContourPoints,
+                            ContourNumber: index + 1,
+                            ContourData
+                        });
+                    });
```

Comment:
I don't think this function should deal with RTSS, it should just create contours (list of points) from given segmentation. And later you convert those in to the desired format by RTSS